src/Components/Modals/AddAdditionalRepositoriesModal.js

```diff
@@ -46,47 +64,49 @@ const AddAdditionalRepositoriesModal = (props) => {
           addSuccessNotification(
             `Repositories have been added for '${keyName}'`
           );
+          setIsSubmitting(false);
         },
         onError: () => {
           addErrorNotification('Something went wrong', {
             description:
               'Your repositories could not be added. Please try again.',
           });
+          setIsSubmitting(false);
+          handleModalToggle();
         },
       }
     );
   };
-
   const editAdditionalRepositoriesDescription =
     'The core repositories for your operating system version, for example BaseOS and AppStream, are always enabled and do not need to be explicitly added to the activation key.';
-
   const editChangesButtons = (
     <ActionGroup>
+      <ProgressButton
+        selectedRepositories={selectedRepositories}
+        keyName={keyName}
+        submitForm={submitForm}
+        isSubmitting={isSubmitting}
+      >
+        Save Changes
```

Comment:
"Save Changes" here isn't used since you populate the text inside the component now